streams/src/test/java/org/apache/kafka/streams/perf/CachingPerformanceBenchmarks.java

```diff
@@ -0,0 +1,359 @@
+        for (int i = 0; i < HALF_MILLION; i++) {
+            LONG_VALS[i] = memoryMeter.measureDeep(new ComplexObject());
+        }
+        long end = System.currentTimeMillis();
+        return end - start;
+    }
+
+    private static long simpleMemoryDeepTrackingTiming() {
+        long start = System.currentTimeMillis();
+        for (int i = 0; i < HALF_MILLION; i++) {
+            LONG_VALS[i] = memoryMeter.measureDeep(STRING_MESSAGE + i);
+        }
+        long end = System.currentTimeMillis();
+        return end - start;
+    }
+
+    private static long complexMemoryTrackingTiming() {
+        long start = System.currentTimeMillis();
+        for (int i = 0; i < HALF_MILLION; i++) {
+            LONG_VALS[i] = memoryMeter.measure(new ComplexObject());
+        }
+        long end = System.currentTimeMillis();
+        return end - start;
+    }
+
+    private static long simpleMemoryTrackingTiming() {
+        long start = System.currentTimeMillis();
+        for (int i = 0; i < HALF_MILLION; i++) {
+            LONG_VALS[i] = memoryMeter.measure(STRING_MESSAGE + i);
+        }
+        long end = System.currentTimeMillis();
+        return end - start;
+    }
+
+    private static long complexSerializationTiming() throws Exception {
+        long start = System.currentTimeMillis();
+        for (int i = 0; i < HALF_MILLION; i++) {
+            byte[] bytes = OBJECT_MAPPER.writeValueAsBytes(new ComplexObject());
+            BYTE_VALS[i] = bytes;
+        }
+        long end = System.currentTimeMillis();
+        return end - start;
+    }
+
+    private static long simpleSerializationTiming() {
+        long start = System.currentTimeMillis();
+        for (int i = 0; i < HALF_MILLION; i++) {
+            byte[] bytes = STRING_SERDE.serializer().serialize("topic", STRING_MESSAGE + i);
+            BYTE_VALS[i] = bytes;
+        }
+        long end = System.currentTimeMillis();
+        return end - start;
+    }
+
+    private static MockMemoryLRUCache<byte[], byte[]> getBytesCache(int size) {
+        return getCache("bytesCache", size, BYTES_SERDE, BYTES_SERDE);
+    }
+
+    private static MockMemoryLRUCache<Integer, String> getIntKeyStringValueCache(int size) {
+        return getCache("objectCache", size, INTEGER_SERDE, STRING_SERDE);
+    }
+
+    private static <K, V> MockMemoryLRUCache<K, V> getCache(String name, int size, Serde<K> keySerde, Serde<V> valueSerde) {
+        return new MockMemoryLRUCache<>(name, size, keySerde, valueSerde);
+    }
+
+    private static <K, V> MockMemoryLRUCache.EldestEntryRemovalListener<K, V> getThrowingRemovalListener() {
+        return new MockMemoryLRUCache.EldestEntryRemovalListener<K, V>() {
+            @Override
+            public void apply(K key, V value) {
+                throw new IllegalStateException("Should be no evictions");
+            }
+        };
+    }
+
+    private static class BlackHole<T> {
+        public  void takeValue(T value) {
+            value = null;
```

Comment:
The JIT can easily inline this method, so it doesn't actually do anything. The BlackHole implementation in JMH is a lot more complex: http://hg.openjdk.java.net/code-tools/jmh/file/cde312963a3d/jmh-core/src/main/java/org/openjdk/jmh/logic/BlackHole.java#l117